qcodes/instrument_drivers/Keysight/N9030B.py

```diff
@@ -0,0 +1,601 @@
+                f"{stop_offset}"
+            )
+
+        if val <= start_offset or abs(val-start_offset) < 10:
+            self.log.warning(f"Provided stop frequency offset {val} Hz was "
+                             f"less than preset start frequency offset "
+                             f"{start_offset} Hz. Provided stop frequency "
+                             f"offset {val} Hz is set and new start freq offset"
+                             f" is: {self.start_offset()} Hz.")
+
+    def _get_data(self, trace_num: int) -> ParamRawDataType:
+        """
+        Gets data from the measurement.
+        """
+        raw_data = self.ask(f":READ:{self.root_instrument.measurement()}{1}?")
+        trace_res_details = np.array(
+            raw_data.rstrip().split(",")
+        ).astype("float64")
+
+        if len(trace_res_details) != 7 or (
+                len(trace_res_details) >= 1 and trace_res_details[0] < -50
+        ):
+            self.log.warning("Carrier(s) Incorrect or Missing!")
+            return -1 * np.ones(self.npts())
+
+        try:
+            data_str = self.ask(f":READ:{self.root_instrument.measurement()}"
+                                f"{trace_num}?")
+            data = np.array(data_str.rstrip().split(",")).astype("float64")
+        except TimeoutError as e:
+            raise RuntimeError("Couldn't receive any data. Command timed "
+                               "out.") from e
+
+        trace_data = data[1::2]
+        return trace_data
+
+    def setup_log_plot_sweep(self,
+                             start_offset: float,
+                             stop_offset: float,
+                             npts: int
+                             ) -> None:
+        """
+        Sets up the Log Plot measurement sweep for Phase Noise Mode.
+        """
+        self.root_instrument.mode("PNOISE")
+        if "LPL" in self.root_instrument._available_meas():
+            self.root_instrument.measurement("LPL")
+        else:
+            raise RuntimeError("Log Plot measurement is not available on your "
+                               "Keysight N9030B instrument with Phase Noise "
+                               "mode.")
+
+        self.start_offset(start_offset)
+        self.stop_offset(stop_offset)
+        self.npts(npts)
+
+    def autotune(self) -> None:
+        """
+        Autotunes frequency
+        """
+        self.write(":SENSe:FREQuency:CARRier:SEARch")
+        self.start_offset()
+        self.stop_offset()
+
+
+class N9030B(VisaInstrument):
+    """
+    Driver for Keysight N9030B PXA signal analyzer. Keysight N9030B PXA
+    siganl analyzer is part of Keysight X-Series Multi-touch Signal
+    Analyzers.
+    This driver allows Swept SA measurements in Spectrum Analyzer mode and
+    Log Plot measurements in Phase Noise mode of the instrument.
+
+    Args:
+        name
+        address
+    """
+
+    CHANNEL_CLASS_1 = SpectrumAnalyzerMode
+    CHANNEL_CLASS_2 = PhaseNoiseMode
```

Comment:
What is the point of these lines? I cannot see that they are ever used. Other instruments do this so that it is possible to replace the channel class in a subclass but for that to work you will need to use these attributes below when creating the modules

Reply:
Yes, these are not used now. I will remove them. Initially, I had used them to create  ChannelLists.